src/widgets/table.rs

```diff
@@ -195,6 +195,34 @@ impl HighlightSpacing {
     }
 }
 
+/// Sets the area of the table that should be highlighted with a selection.
+#[derive(Debug, Display, EnumString, PartialEq, Eq, Clone, Default, Hash)]
+pub enum HighlightArea {
+    /// No highlight
+    None,
+    /// Highlight cell only
+    Cell,
+    /// Highlight entire row
+    #[default]
+    Row,
+    /// Highlight entire column
+    Col,
+    /// Highlight entire row and column
+    RowAndCol,
+}
```

Comment:
I think this is a good idea, as it is better for the Table widget to be more customizable rather than less. The only thing we should decide is the order in which we apply highlights. I'm not really sure if we can leave this decision to the user because they would set any highlight first, and then call `render_statefull_widget`, so by the time we get to actually highlight rows, columns and cells, we cannot trace the order? (Correct me if I'm wrong)

What would happen if I do this actions:
1. First, Highlight row with red background
2. Second, Highlight column with blue background

Would the intersection of the row and column highlights should be red or blue? Maybe the order should be from "more visible" to "less visible":
1. Cell
2. Row
3. Col
so the intersection would be red, even if the column highlight was applied later by the user.

On the other hand, if we wanted to give the user a choice, at the top of my head I think we could change :
```rust
struct Table{
..
highlight_style: Style
}
``` 
to something like:

```rust
enum TableHighlightStyle {
   Row(Style),
   Col(Style),
   Cell(Style),
}

struct Table { 
..
    highlight_style: Vec<TableHighlightStyle>,
}
```
We can then use the vec order to apply each highlight.

I think we should only expose individual methods that use `Style` and keep the `TableHighlightStyle` enum private:

```rust
// add specific functions for each type of highlight.
fn highlight_row(style: Style);
fn highlight_col(style: Style);
fn highlight_cell(style: Style);


// Optionally, we could deprecate the original method (similar as what you suggested with selection() 
// to avoid breaking changes),
#[deprecated]
fn highlight_table(style: Style);

```
To me, the second option could be a lot more interesting

Reply:
3 functions seems nice to me. The current `highlight` function is already the equivalent of `highlight_row`, so we can keep it with a deprecated attribute, and replace the implementation with `self.highlight_row()`.

If we apply styles row first then column then cell, does putting cell last effectively allow any problems introduced by the column to be undone by the cell? If so does that avoid having to worry about exposing the order to the caller (and introducing more complexity)?

Reply:
> 3 functions seems nice to me. The current `highlight` function is already the equivalent of `highlight_row`, so we can keep it with a deprecated attribute, and replace the implementation with `self.highlight_row()`.

I think this would be great!


> If we apply styles row first then column then cell, does putting cell last effectively allow any problems introduced by the column to be undone by the cell? If so does that avoid having to worry about exposing the order to the caller (and introducing more complexity)?

I'm not quite sure if I understand correctly... If the user does this:
1. Highlight row with red background
2. Highlight column with blue background 
3. Highlight cell with green background

Effectively it is as you say... you solve the problem, as the cell is the intersection of row and col. So the row would be red, the column blue, and the cell green. All good. 

But what if the user doesn't highlight the cell?
1. Highlight row with red background
2. Highlight column with blue background 

What color should the cell be? We can:
a) Color in an arbitrary order
b) Force the user to always color cell
c) Give the user a way to define the order

Option b seems like the worst, and I don't think is viable. Option C could be better, but I genuinely don't know if the "hidden behaviour" would be ok (The user would have to know that the last highlight is the one who would be visible). Also, Option A feels like it will also have some hidden behaviour, as the order in which we color would be hidden from the end user.

Reply:
I'm thinking about how would the Option B behaviour translates to other widgets.. and I think that 'Last style applied is visible' is similar to what ratatui already does when applying multiple Styles. 

For example, if you do:

```rust
Paragraph::new("Ipsum Lorem")
.style(Style::default().fg(Color::Cyan).bg(Color::Black))
.style(Style::default().fg(Color::Red).bg(Color::Blue));
```

The second line is effectively ignored, because the third line sets the same parameters again. This way, the final fg color would be Red, and the bg color would be Blue. So, in a sense, it always works like painting a wall, if you paint it first white, then paint it red, the wall will be red.

Reply:
>But what if the user doesn't highlight the cell?
>
>  1.  Highlight row with red background
>  2. Highlight column with blue background

Then they are choosing to accept the order that we render (row first, column next). If they want the opposite then they can easily add 3. highlight cell with red background. By being opinionated we don't make it impossible to achieve whatever order is needed.

Row first is not really that arbitrary, because we're moving from an existing implementation where only row highlighting was available to one where other highlighting is also available. Cell must go last as otherwise it's useless whenever either or Row or Column highlighting is set, so Row/Column order really matters. Given the new Cell behavior is rendered on top of the Row, it would seem weird (IMO) to render the new column highlight behind the Row (when considering the styles as overwriting the styles behind or underneath)

So Row/Column/Cell makes the most sense here.

Reply:
> By being opinionated we don't make it impossible to achieve whatever order is needed.
This is a good stance. I like it!

> So Row/Column/Cell makes the most sense here.
Got it. It is simple enough.. We just have to document it.

Reply:
Done